service/core/kafka/server/src/main/scala/com/lightbend/lagom/internal/broker/kafka/Producer.scala

```diff
@@ -84,15 +85,18 @@ private[lagom] object Producer {
     override def receive = {
       case EnsureActive(tag) =>
         val daoFuture = offsetStore.prepare(s"topicProducer-$topicId", tag)
-        // Left means no service lookup was attempted, Right means a service lookup was done - this allows us to
-        // distinguish between no service lookup, and a not found service lookup.
-        val serviceUriFuture: Future[Either[Unit, Option[URI]]] = kafkaConfig.serviceName match {
```

Comment:
Yes, I had a commit with a `ServiceLookup` ADT. Later I realised that I could remove it by also resolving it much earlier in the stack. 

The only drawback is that later, when logging failures, we have to do some branching using the service name.